pkg/bitcoin_difficulty/chain.go

```diff
@@ -0,0 +1,34 @@
+package bitcoin_difficulty
```

Comment:
From https://go.dev/blog/package-names:

> Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns (...)
>
> The style of names typical of another language might not be idiomatic in a Go program. Here are two examples of names that might be good style in other languages but do not fit well in Go:
> - computeServiceClient
> - priority_queue

That being said, what do you think about naming this package `btcdiff`? (`btcdifficulty` feels too long/complicated in my opinion).

Reply:
Makes sense. Done in 3ec575c469ee0d1dc37503cd59cae39660cbfa11.

Reply:
can we also change `pkg/bitcoin_difficulty` -> `pkg/btcdiff`?

Reply:
Sure, changed package name to `btcdiff` in 3ff897098673c429bc9696f78cfcbde217d814cb.